x509util/utils_test.go

```diff
@@ -176,6 +182,38 @@ func Test_generateSubjectKeyID(t *testing.T) {
 	}
 }
 
+func Test_generateSubjectKeyID_fips(t *testing.T) {
+	if !fipsutil.Enabled() {
+		t.Skip("FIPS 140-3 mode is not enabled")
+	}
+
+	sha256Cert := decodeCertificateFile(t, "testdata/letsencrypt.crt")
+	type args struct {
+		pub crypto.PublicKey
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    []byte
+		wantErr bool
+	}{
+		{"ok", args{sha256Cert.PublicKey}, sha256Cert.SubjectKeyId, false},
+		{"fail", args{[]byte("fail")}, nil, true},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := generateSubjectKeyID(tt.args.pub)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("generateSubjectKeyID() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("generateSubjectKeyID() = %v, want %v", got, tt.want)
+			}
```

Comment:
You could use `require` here. It's used in some other places in `crypto` already.